lib/knex-builder/internal/config-resolver.js

```diff
@@ -42,9 +40,11 @@ function resolveConfig(config) {
 
   // If config connection parameter is passed as string, try to parse it
   if (typeof parsedConfig.connection === 'string') {
-    resolvedConfig = Object.assign({}, parsedConfig, {
-      connection: parseConnection(parsedConfig.connection).connection,
-    });
+    resolvedConfig = Object.assign(
+      {},
+      parsedConfig,
+      parseConnection(parsedConfig.connection)
```

Comment:
`parseConnection().client` goes into `Dialect` variable in previous code fragments